.devcontainer/Dockerfile

```diff
@@ -0,0 +1,39 @@
+# --------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------
+# Dockerfile to Develop inside Dev Container
+
+ARG VARIANT=ubuntu-20.04
+FROM mcr.microsoft.com/vscode/devcontainers/base:0-${VARIANT}
```

Comment:
is this image accessible by non-Microsoft folks?

Reply:
Yes - [Microsoft Container Registry](https://github.com/microsoft/ContainerRegistry) is a public registry - and this is the same registry / image used in the vs code remote extension's predefined devcontainer registration.

To confirm, I was able to pull this image from MCR without logging in.

If it turns out that access is an issue, we could switch to pull from other public registry (e.g. official ubuntu image)

Reply:
"I was able to pull this image from MCR without logging in." It is not enough. This one might be ok, but the next time you need to check if the owner of the image has made the image public. i.e. you can find URL somewhere in a public place.  MCR images are tricky.

Reply:
Microsoft was pushing all pipelines use MCR images due to docker issue (throttling I think). @snnn - any concerns with using ubuntu 20.04 directly (pull from docker)? I think it would be fine since codespace creation happens way less then pipeline runs.

@jamill - do you recall the reasons for the push to MCR? If it is not security or legal then we can change to ubuntu.

Reply:
I chose this image as it is the same image used in the docker file created by VS Code when it sets up a dev container using [predefined configuration file](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/introduction-to-dev-containers#using-a-predefined-dev-container-configuration). The dockerfile for the ubuntu based dev container is in the [vscode-dev-containers repository](https://github.com/microsoft/vscode-dev-containers/blob/main/containers/ubuntu/.devcontainer/Dockerfile).

This image contains some extra common tooling on top of the base ubuntu image.

If there is still concerns about using this image, let me know and I can base it off of the base ubuntu image, but I think given the above reason, it is safe.

Reply:
This one looks good. It is publicly visible from https://github.com/microsoft/vscode-dev-containers/.